client/src/pages/Keyboard.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { FC, useEffect, useRef, useState } from 'react'
+import { PageWrapper } from "../components/PageWrapper";
+import { TextInput } from "../components/TextInput";
+import randomWords from "random-words";
+import classes from './keyboard.module.scss';
+import { Button } from "../components/Button";
+import { COPY_LABEL, GO_HOME_LABEL } from "../utils/constants";
+import { useTimer } from "../components/Timer/hooks/useTimer";
+import { Timer } from "../components/Timer";
+import { observer } from "mobx-react-lite";
+import { useWebsocket } from "./hooks/useWebsocket";
+import appStore from "../stores/appStore";
+import { DisplayTimer } from "../components/GameDetails/DisplayTimer";
+import { ButtonTitle } from "../components/Button/const";
+import { copy } from "../utils/utils";
+import { useNavigate } from "react-router-dom";
+
+export const Keyboard: FC = observer(() => {
+    const {
+        data,
+        roomUsers,
+        gameTime,
+        socket,
+    } = useWebsocket(appStore);
+    const { roomLimit } = appStore.user;
+    const { startTimer, leftTime, isTimerFinished, isTimerStarted } = useTimer({ ms: gameTime * 1000 });
+    const {
+        startTimer: startPreTimer,
+        leftTime: leftPreTime,
+        isTimerFinished: isPreTimerFinished,
+        isTimerStarted: isPreTimerStarted,
+    } = useTimer({ ms: 5000 });
+
+    const [text, setText] = useState('');
+    const [countOfWords, setCountOfWords] = useState(0);
+    const [textPart, setTextPart] = useState({
+        written: '',
+        toWrite: text,
+    });
+
+    const navigate = useNavigate();
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const textInputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+        if(roomLimit && +roomLimit === roomUsers.length) {
+            startPreTimer();
+        }
+    }, [roomUsers.length, roomLimit]);
+
+    useEffect(() => {
+        if(leftPreTime === 0) {
+            startTimer();
+        }
+    }, [leftPreTime]);
+
+    useEffect(() => {
+        setNewWord();
+        if(isTimerStarted) {
+            textInputRef.current?.focus();
+        }
+        if(isTimerFinished) {
+            buttonRef.current?.focus();
+        }
```

Comment:
Also, it will be better to move these calls to separate useEffect with a dependency of `newWord`.
```suggestion
            isTimerFinished && textInputRef.current?.focus();
            isTimerFinished && buttonRef.current?.focus();
```